src/Terminal.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE ImportQualifiedPost #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+module Terminal (terminalGame) where
+
+import Control.Applicative (liftA2)
+import Control.Monad ((<=<))
+import Data.Either.Extra (maybeToEither)
+import Data.Function (on)
+import Data.Map.Strict (empty, findWithDefault)
+import Data.Text qualified as Text
+import Data.Text.IO qualified as TextIO
+import GameLogic (Board (..), Coordinates, Player (..), nextTurn, playAtSpot, winner)
+import System.Console.ANSI (Color (..), ColorIntensity (..), ConsoleLayer (..), SGR (..), clearScreen, setCursorPosition, setSGR)
+import Text.Read (readMaybe)
+
+showPlayerT :: Player -> Text.Text
+showPlayerT = Text.pack . show
+
+showBoard :: Board -> Text.Text
+showBoard (Board board) =
+  "  0   1   2 \n"
+    <> Text.intercalate "\n ---+---+---\n" rows
+  where
```

Comment:
You know, arguably, the best part about `where` is that it lets you focus on the big picture first.

If you just use this construct without much thought you make your code much harder to grasp to others, because you effectively force them to read sections in reverse.

This function, for instance, pretty much requires the reader to read from bottom upwards to make sense, which is almost hostile to other readers.